pkg/operator/starter.go

```diff
@@ -0,0 +1,95 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/cert-manager-operator/pkg/controller/deployment"
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+	"github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/status"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	TargetNamespace   = "openshift-cert-manager"
+	OperatorNamespace = "openshift-cert-manager-operator"
+)
+
+func RunOperator(ctx context.Context, cc *controllercmd.ControllerContext) error {
+	kubeClient, err := kubernetes.NewForConfig(cc.ProtoKubeConfig)
+	if err != nil {
+		return err
+	}
+
+	configClient, err := configv1client.NewForConfig(cc.KubeConfig)
+	if err != nil {
+		return err
+	}
+
+	genericOperatorClient, dynamicInformers, err := genericoperatorclient.NewClusterScopedOperatorClient(
+		cc.KubeConfig, operatorv1.GroupVersion.WithResource("certmanagers"))
+	if err != nil {
+		return err
+	}
+
+	clusterOperator, err := configClient.ConfigV1().ClusterOperators().Get(ctx, "cert-manager-operator", metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return err
+	}
+
+	versionRecorder := status.NewVersionGetter()
```

Comment:
In the CAO I added a comment explaining why the `SetVersion()` is called here because it took me some digging to realize what's going on. Maybe you'd like to include it as well.

https://github.com/openshift/cluster-authentication-operator/blob/ca4d280f750ed8f04bbee47422e6e29da9064d0c/pkg/operator/starter.go#L154-L156

Reply:
added